src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp

```diff
@@ -395,6 +396,88 @@ struct KVAxesPosition {
+        auto res = opp::wrap_type<ov::op::v0::Result>({last_op->output(0)});
+
+        auto callback = [=](ov::pass::pattern::Matcher& m) {
+            auto& node_to_output = m.get_pattern_value_map();
+    
+            auto matched_node_slice = node_to_output.at(slice).get_node_shared_ptr();
+            auto matched_node_matmul = node_to_output.at(matmul).get_node_shared_ptr();
+            std::shared_ptr<ov::Node> matched_node_last_op = nullptr;
+            if (node_to_output.count(matmul_add)) {
+                matched_node_last_op = node_to_output.at(matmul_add).get_node_shared_ptr();
+            } else if (node_to_output.count(matmul_transpose)) {
+                matched_node_last_op = node_to_output.at(matmul_transpose).get_node_shared_ptr();
+            } else if (node_to_output.count(matmul_multiply)) {
+                matched_node_last_op = node_to_output.at(matmul_multiply).get_node_shared_ptr();
+            } else {
+                matched_node_last_op = matched_node_matmul;
+            }
+            auto matched_node_result = node_to_output.at(res).get_node_shared_ptr();
+
+            auto matched_slice = std::static_pointer_cast<ov::op::v8::Slice>(matched_node_slice);
+            auto matched_matmul = std::static_pointer_cast<ov::op::v0::MatMul>(matched_node_matmul);
+            auto matched_result = std::static_pointer_cast<ov::op::v0::Result>(matched_node_result);
+
+            // Cut original model after Slice Op:
+            matched_result->input(0).replace_source_output(matched_slice);
+            matched_result->output(0).set_names({"output_embed"});
+            matched_result->validate_and_infer_types();
+
+            // Create an additional model after Slice Op:
+            auto new_param = std::make_shared<ov::op::v0::Parameter>(matched_slice->output(0).get_element_type(),
+                                                                     matched_slice->output(0).get_partial_shape());
+            new_param->output(0).add_names({"output_embed"});
+            matched_matmul->input(0).replace_source_output(new_param);
+            auto new_result = std::make_shared<ov::op::v0::Result>(matched_node_last_op);
+            tail_vocab_mm_ref.get() = std::make_shared<ov::Model>(
+                ov::OutputVector{new_result->output(0)},
+                ov::ParameterVector{new_param},
+                "TailVocabMatmul");
+           
+            return true;
+        };
+        register_matcher(std::make_shared<opp::Matcher>(res, "CutTailAfterSlice"), std::move(callback));
+
+    }
+};
+
+namespace {
+std::shared_ptr<ov::Model> cut_tail_after_slice(std::shared_ptr<ov::Model>& model) {
+    ov::pass::GraphRewrite rewr;
+    std::shared_ptr<ov::Model> tail_vocab_mm = nullptr;
+    std::reference_wrapper<std::shared_ptr<ov::Model>> tail_vocab_mm_ref(tail_vocab_mm);
+    rewr.add_matcher<CutTailAfterSlice>(tail_vocab_mm_ref);
+    rewr.run_on_model(model);
```

Comment:
shouldn't `validate_nodes_and_infer_types()` be called on the `model` right here?